src/libraries/System.Collections/src/System/Collections/Generic/OrderedDictionary.cs

```diff
@@ -392,8 +392,9 @@ public TValue this[TKey key]
         /// - OverwriteExisting: If the key already exists, overwrites its value with the specified value, e.g. this[key] = value
         /// - ThrowOnExisting: If the key already exists, throws an exception, e.g. Add(key, value)
         /// </param>
+        /// <param name="keyIndex">The index of the added or existing key. This is always a valid index into the dictionary.</param>
         /// <returns>true if the collection was updated; otherwise, false.</returns>
-        private bool TryInsert(int index, TKey key, TValue value, InsertionBehavior behavior)
+        private bool TryInsert(int index, TKey key, TValue value, InsertionBehavior behavior, out int keyIndex)
```

Comment:
Given that there already is an `index` parameter representing the same concept, would it make sense to fuse the two parameters into one and have it being passed by reference?

Reply:
I'm actually more inclined to have it be the separate out, primarily because there are lots of accesses to `index`, each of which will become an indirection if it's made into a ref.

Reply:
I think as a helper method the 4-parameter signature is pretty clean considering the use sites:
```cs
TryInsert(index, key, value, InsertionBehavior.ThrowOnExisting, out _);
bool modified = TryInsert(index: -1, key, value, InsertionBehavior.OverwriteExisting, out _);
TryInsert(index: -1, key, value, InsertionBehavior.ThrowOnExisting, out _);
return TryInsert(index: -1, key, value, InsertionBehavior.IgnoreInsertion, out index);
```
It allows us to pass constant ints as the index and ignore the out parameter if unnecessary. Is your concern due to something else like perf?

Reply:
No concerns other than duplication. We can keep as-is.